macaroon.js

```diff
@@ -7,14 +7,27 @@ nacl.util = require('tweetnacl-util');
 const NONCELEN = 24;
 
 /**
-  Check if supplied value is a string. Throws if not.
+  Check if supplied value is a string and return it. Throws if not.
   @param {String} val The value to assert as a string
   @param {String} msg The value label.
 */
 function requireString(val, msg) {
   if (typeof val !== 'string') {
     throw new Error(`${msg}, is not of type string.`);;
   }
+  return val
+}
+
+/**
+  Check if supplied value is a string or undefined or null. Throws
+  if not. Always returns a string (the empty string if undefined
+  or null).
+*/
```

Comment:
This is missing JSDoc param & return values.

Reply:
done